src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/CryptoConfig.cs

```diff
@@ -640,5 +389,265 @@ private static void EncodeSingleOidNum(uint value, byte[] destination, ref int i
+            // Enhanced Key Usage OID value
+            nameToFullTypeName.Add("2.5.29.37", "System.Security.Cryptography.X509Certificates.X509EnhancedKeyUsageExtension, " + AssemblyName_X509Certificates);
+
+            // X509Chain class can be overridden to use a different chain engine.
+            nameToFullTypeName.Add("X509Chain", "System.Security.Cryptography.X509Certificates.X509Chain, " + AssemblyName_X509Certificates);
+
+            // PKCS9 attributes
+            nameToFullTypeName.Add("1.2.840.113549.1.9.3", "System.Security.Cryptography.Pkcs.Pkcs9ContentType, " + AssemblyName_Pkcs);
+            nameToFullTypeName.Add("1.2.840.113549.1.9.4", "System.Security.Cryptography.Pkcs.Pkcs9MessageDigest, " + AssemblyName_Pkcs);
+            nameToFullTypeName.Add("1.2.840.113549.1.9.5", "System.Security.Cryptography.Pkcs.Pkcs9SigningTime, " + AssemblyName_Pkcs);
+            nameToFullTypeName.Add("1.3.6.1.4.1.311.88.2.1", "System.Security.Cryptography.Pkcs.Pkcs9DocumentName, " + AssemblyName_Pkcs);
+            nameToFullTypeName.Add("1.3.6.1.4.1.311.88.2.2", "System.Security.Cryptography.Pkcs.Pkcs9DocumentDescription, " + AssemblyName_Pkcs);
+
+            return nameToFullTypeName;
+
+            // Types in Desktop but currently unsupported in CoreFx:
+            // string RIPEMD160ManagedType = "System.Security.Cryptography.RIPEMD160Managed" + AssemblyName_Encoding;
+            // string ECDiffieHellmanCngType = "System.Security.Cryptography.ECDiffieHellmanCng, " + AssemblyName_Cng;
+            // string MD5CngType = "System.Security.Cryptography.MD5Cng, " + AssemblyName_Cng;
+            // string SHA1CngType = "System.Security.Cryptography.SHA1Cng, " + AssemblyName_Cng;
+            // string SHA256CngType = "System.Security.Cryptography.SHA256Cng, " + AssemblyName_Cng;
+            // string SHA384CngType = "System.Security.Cryptography.SHA384Cng, " + AssemblyName_Cng;
+            // string SHA512CngType = "System.Security.Cryptography.SHA512Cng, " + AssemblyName_Cng;
+            // string SHA256CryptoServiceProviderType = "System.Security.Cryptography.SHA256CryptoServiceProvider, " + AssemblyName_Csp;
+            // string SHA384CryptoSerivceProviderType = "System.Security.Cryptography.SHA384CryptoServiceProvider, " + AssemblyName_Csp;
+            // string SHA512CryptoServiceProviderType = "System.Security.Cryptography.SHA512CryptoServiceProvider, " + AssemblyName_Csp;
+            // string DpapiDataProtectorType = "System.Security.Cryptography.DpapiDataProtector, " + AssemblyRef.SystemSecurity;
+        }
+
+        private static ConcurrentDictionary<string, Type> CreateTypeNameToTypeMapping()
+        {
+            ConcurrentDictionary<string, Type> typeNameToType = new ConcurrentDictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
+
+            Type HMACMD5Type = typeof(System.Security.Cryptography.HMACMD5);
```

Comment:
It is better to get the type right next to its use. Fetching all types first into locals will result in larger stack frame and bigger native code.